lobster_common/Vec3.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+
+import numbers
+from typing import Union, List, Tuple
+
+import numpy as np
+
+from lobster_common import Quaternion
+
+from lobster_common.exceptions import InputDimensionError
+
+
+class Vec3:
+    """
+    Data class that stores 3 dimensional vectors. The vectors should always be stored in the NED coordinate system, if
+    if there ar
+    """
+
+    def __init__(self, data: Union[List[float], Tuple[float, float, float], np.ndarray, 'Vec3']):
+        """
+        Creates a 3 dimensional vector from a data array
+        :param data: Array with length 4 in the form [x, y, z]
+        """
+        assert isinstance(data, np.ndarray) or isinstance(data, List) or isinstance(data, Tuple) or isinstance(data, Vec3)
+
+        if isinstance(data, Vec3):
+            self._data: np.ndarray = np.asarray(data.numpy().copy())
+        else:
+            self._data: np.ndarray = np.asarray(data)
+
+        assert self._data.shape
+        if self._data.shape[0] != 3:
+            raise InputDimensionError("A Vec3 needs an input array of length 3")
+        elif self._data.dtype != float and self._data.dtype != int:
+            raise TypeError(
+                f"A Vec3 needs to be instantiated by an array of floats, not an array of {self._data.dtype}")
+
+    def asENU(self) -> np.ndarray:
+        """
+        Transforms the vector to the ENU coordinate system.
+        :return: Vector in the ENU coordinate system.
+        """
+        # Swapping the Y and Z axes
+        array = self._data.copy()
+        array[1] = -array[1]
+        array[2] = -array[2]
+        return array
```

Comment:
You don't have to copy the array first:
```suggestion
        return np.array([array[0],-array[1],-array[2]])
```

Reply:
This conversation was resolved but I didn't got any feedback from you why?

Reply:
Oh because I just pressed commit suggestion